File: app/src/main/java/com/revosleap/bxplayer/AppUtils/Player/PlayerAdapter.java
package com.revosleap.bxplayer.AppUtils.Player;

import android.app.Activity;
import android.media.MediaPlayer;
import android.support.annotation.NonNull;

import com.revosleap.bxplayer.AppUtils.Models.Album;
import com.revosleap.bxplayer.AppUtils.Models.Song;

import java.util.List;

public interface PlayerAdapter {
    void initMediaPlayer();

    void release();

    boolean isMediaPlayer();

    boolean isPlaying();

    void resumeOrPause();

    void reset();

    boolean isReset();

    void instantReset();

    void skip(final boolean isNext);

    void openEqualizer(@NonNull final Activity activity);

    void seekTo(final int position);

    void setPlaybackInfoListener(final PlaybackInfoListener playbackInfoListener);

    Song getCurrentSong();

    @PlaybackInfoListener.State
    int getState();

    int getPlayerPosition();

    void registerNotificationActionsReceiver(final boolean isRegister);

    Album getSelectedAlbum();

    void setSelectedAlbum(final Album album);

    void setCurrentSong(@NonNull final Song song, @NonNull final List<Song> songs);

    MediaPlayer getMediaPlayer();

    void onPauseActivity();

    void onResumeActivity();
}

File: app/src/main/java/com/revosleap/bxplayer/callbacks/PlayerAdapter.kt
package com.revosleap.bxplayer.callbacks


import android.app.Activity
import android.media.MediaPlayer
import com.revosleap.bxplayer.models.Album
import com.revosleap.bxplayer.models.Song
import com.revosleap.bxplayer.utils.playback.PlaybackInfoListener


interface PlayerAdapter {

    fun isMediaPlayer(): Boolean

    fun isPlaying(): Boolean

    fun isReset(): Boolean

    fun getCurrentSong(): Song?

    fun getCurrentSongs():MutableList<Song>?

    fun getPlayerPosition(): Int

    fun getSelectedAlbum(): Album?

    fun initMediaPlayer()

    fun shufflePlayList()

    fun release()

    fun resumeOrPause()

    fun reset()

    fun instantReset()

    fun skip(isNext: Boolean)

    fun openEqualizer(activity: Activity)

    fun seekTo(position: Int)

    fun setPlaybackInfoListener(playbackInfoListener: PlaybackInfoListener)

    fun registerNotificationActionsReceiver(isRegister: Boolean)

    fun setCurrentSong(song: Song, songs: MutableList<Song>)

    fun onPauseActivity()

    fun onResumeActivity()

    fun setSelectedAlbum(album: Album)

    fun getMediaPlayer(): MediaPlayer?

    @PlaybackInfoListener.State
    fun getState(): Int
}
